server/datastore/mysql/migrations/tables/20250410104321_UpdateMacOSSoftwareNames.go

```diff
@@ -76,38 +85,49 @@ WHERE
 		WHERE
 			hs2.software_id = ?
 			AND hs2.host_id = hs1.host_id)`
+	updateHostSoftwareInstalledPathsStmt := `UPDATE host_software_installed_paths SET software_id = ? WHERE software_id IN (?)`
 
-	var allExcludedIDs []uint
-	for bid, excluded := range excludedIDs {
-		allExcludedIDs = append(allExcludedIDs, excluded...)
-		var hs []struct {
+	var allExcludedIDs []uint64
+	for newChecksum, idsToMerge := range idsToMergeByNewChecksum {
+		allExcludedIDs = append(allExcludedIDs, idsToMerge...)
+		var hostIDRecordList []struct {
 			HostID uint `db:"host_id"`
 		}
-		selectedID, ok := selectedIDs[bid]
+		selectedID, ok := selectedIDs[newChecksum]
 		if !ok {
-			return fmt.Errorf("%s had excluded IDs but no selected ID", bid)
+			return fmt.Errorf("%v excluded IDs but no selected ID", idsToMerge)
 		}
 
-		stmt, args, err := sqlx.In(getRecordToUpdateStmt, excluded, selectedID)
+		stmt, args, err := sqlx.In(getRecordToUpdateStmt, idsToMerge, selectedID)
 		if err != nil {
-			return fmt.Errorf("sqlx.In for getting host software records to update for bundle_id %s: %w", bid, err)
+			return fmt.Errorf("sqlx.In for getting host software records to update for old software IDs %v: %w", idsToMerge, err)
 		}
 
-		if err := txx.Select(&hs, stmt, args...); err != nil {
+		if err := txx.Select(&hostIDRecordList, stmt, args...); err != nil {
 			if errors.Is(err, sql.ErrNoRows) {
 				// if there are no rows, this means the host is already pointed at the selected software
 				// ID, so no update needed
 				continue
 			}
-			return fmt.Errorf("getting host software record to update for bundle_id %s: %w", bid, err)
+			return fmt.Errorf("getting host software record to update for old software IDs %v: %w", idsToMerge, err)
 		}
 
-		for _, h := range hs {
+		for _, h := range hostIDRecordList {
 			_, err = tx.Exec(`INSERT INTO host_software (host_id, software_id) VALUES (?, ?)`, h.HostID, selectedID)
 			if err != nil {
-				return fmt.Errorf("updating host_software.software_id for bundle_id %s: %w", bid, err)
+				return fmt.Errorf("updating host_software.software_id for old software IDs %v: %w", idsToMerge, err)
 			}
 		}
```

Comment:
Insert rows with the "merged" software ID into `host_software` for any hosts that need it but don't already have it (as calculated by `getRecordToUpdateStmt()`), and add the merged IDs to a cumulative list of IDs `allExcludedIDs` that we can delete software rows for when we're done.

Reply:
Yep, the add-to-exclude is in L92.